### ISSUE TITLE
Deletion of subnets will try to delete non-managed EIPs connected to non-managed Egress: nat-* gateways

### ISSUE THREAD

**Issue by jhohertz** [kubernetes/kops#7187]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

1.12.1

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

1.12.9

**3. What cloud provider are you using?**

AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

kops delete cluster mycluster.name --yes

**5. What happened after the commands executed?**

It will loop continuously once it tries to delete EIPs it should not be attempting to, failing each time. If interrupted once the rest of the resources are removed, and re-run, it will correctly identify there is nothing left to delete and complete.

While trying to delete the EIPs, there are many lines like these emitted:

```elastic-ip:eipalloc-0b4dbe528cb2bd659	error deleting resources, will retry: error deleting elastic ip "1.234.56.78": AuthFailure: You do not have permission to access the specified resource.```

**6. What did you expect to happen?**

The NAT gateway in question that these EIPs are bound to, are NOT managed by kops, rather they are specified in the cluster spec as `Egress: nat-abcd1245` as pre-created, shared NATs.

It should not be attempting to delete these IPs. Notably it does not appear to attempt to delete the NAT itself, so why it's trying to delete it's EIPs is... interesting.

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

I can sent the whole thing if needed, but for now will just show a relevant snip... all the private subnets are specced in this manner:

```
  subnets:
  - cidr: 10.5.47.0/24
    name: us-east-1f
    type: Private
    zone: us-east-1f
    egress: nat-0a12f44ee28f57012
```

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

Will do so on request if needed. Relevant snippets above.

**9. Anything else do we need to know?**

Don't think so.

**Comment by jhohertz** [comment]
This looks like it might be addressed by #7525, will close once I confirm.

**Comment by jhohertz** [comment]
Confirmed fixed in 1.14.0, closing.
